### PR TITLE
Use string Version type for ApplicationInstallation CRD

### PR DESCRIPTION
**Pull request by rastislavs**
Signed-off-by: Rastislav Szabo <rastislav@kubermatic.com>

**What this PR does / why we need it**:
The usage of `Masterminds/semver` in our `ApplicationInstallation` CRD causes troubles by reconciling it using our reconciliation packages, as it keeps data in unexported fields which causes panic in `DeepEqual` comparison. It looks like using the string type + validation pattern for the Version in the CRD and using semver libs just internally in the validation / implementation logic is a better approach.

Even though this is an API change, it is not breaking as the type in the generated OpenAPIV3Schema remains the same - string.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind cleanup
/kind api-change

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Use string Version type for ApplicationInstallation CRD.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -625,7 +624,7 @@ type ApplicationRef struct {
 	Name string `json:"name" required:"true"`
 
 	// Version of the Application. Must be a valid SemVer version
```
Comment (SimonTheLeg) [inline pkg/api/v1/types.go:626]: let's add a dev comment above the field, that explains why we chose a string for now and links to the split semver PR

**Comment by SimonTheLeg** [comment]
Just a small doc request from my side. This PR looks good to me. In addition I think we need a PR in dashboard, as some of types are being used there (e.g. the conversion funcs).

**Review by vgramer — APPROVED** [review]
/lgtm

**Review by vgramer — APPROVED** [review]
/lgtm

**Comment by SimonTheLeg** [comment]
🥳

**Comment by embik** [comment]
@rastislavs could this be considered a bug fix? It seems that maybe 2.21 can encounter a reconcile loop, judging by the description. Do we need to backport this?

**Comment by rastislavs** [comment]
> @rastislavs could this be considered a bug fix? It seems that maybe 2.21 can encounter a reconcile loop, judging by the description. Do we need to backport this?

I don't think so it is a bug per say, just a limitation. In 2.21 we are not trying to reconcile the ApplicationInstallation by any of our controllers (that would come with a new feature in the next KKP release). So it should not cause any troubles in 2.21.

**Pull request merged**